server/controllers/availability.js

```diff
@@ -0,0 +1,99 @@
+const Availability = require("../models/availability");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+// @route POST /schedule/
+// @desc Create a schedule
+// @access Public
+exports.createSchedule = asyncHandler(async (req, res, next) => {
+  const profileId = req.profile.id;
+  const { name, days } = req.body;
+  const schedule = await Availability.findOne({ name, petSitterId: req.profile.id });
+  if (schedule) {
+    res.status(400);
+    throw new Error("Schedule name already exist");
+  }
+  const newSchedule = new Availability.create({
+    profileId,
+    name,
+    days,
+  });
+  if (newSchedule) {
+    if (!req.profile.activeSchedule) {
+      req.profile.set({ activeSchedule: newSchedule.id });
+      await req.profile.save();
+    }
+    res.status(201).json({
+      success: {
+        newSchedule
+      }
+    });
+  } else {
```

Comment:
I don't think we'll reach this else block if the above async operation fails, the `asyncHandler` will instead forward the error to our error middleware

Ditto where applicable

Reply:
Done the changes